### PR TITLE
LPD-15236 Use a lowercase object definition name for the JAX-RS properties and update the usages of the getOSGiJaxRsName method

### PR DESCRIPTION
**Pull request by ccorreagg**
Related ticket: 

---

The purpose of this ticket is to fix the bug of the related ticket. In order to reproduce the bug, please, see the related ticket.

In this PR, we have performed the following changes:

- The `ObjectDefinitionImpl.getOSGiJaxRsName` method has been updated to use the lowercase name instead of just the name. This has been done because, as two objects with similar names (for example `Test` and `test`) shares the same JAX-RS Application path (`/c/tests`), we need to set the JAX-RS related properties also considering the lowercase keys.
- As Batch Engine needs a taskItemDelegateName property to identify which object definition is importable/exportable, we have updated the `ObjectDefinitionDeployerImpl` class of the Objects REST API IMPL module to deploy one `VulcanBatchEngineTaskItemDelegate` implementation (that is: ObjectEntryResourceImpl) for each Object Definition, with the properties such as the companyId and the object definition name to avoid confussions with other objects with similar name in different companies.
- Update the usages of the `getOSGiJaxRsName` method to avoid using it if it is not JAX- RS related.
- Update the usages of the properties `entity.class.name` and `batch.engine.entity.class.name` to use them when it applies

### CODE REVIEW

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:stop

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Review by magjed4289 — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:stable

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java [hunk 1]
```diff
@@ -138,6 +155,106 @@ public void tearDown() {
...
+					"className", "com.liferay.object.rest.dto.v1_0.ObjectEntry"
+				).put(
+					"errorMessage", ""
+				).put(
+					"executeStatus", "COMPLETED"
+				).toString(),
+				jsonObject3.toString(), JSONCompareMode.LENIENT);
+		}
+		catch (AssertionError error) {
+			Assert.fail(jsonObject3.getString("errorMessage"));
```
Comment (ccorreagg) [inline modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java:219]: [echo] PMD VIOLATION: /opt/dev/projects/github/liferay-portal/modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java:219 > Do not call Assert#fail improperly. (PMD Rule: AssertFailJUnitRule)

Comment (ccorreagg) [reply modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java:219]: Could you check @magjed4289 ?

Comment (magjed4289) [reply modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java]: I have removed the try catch block

**Comment by ccorreagg** [comment]
ci:forward

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:forward:force

**Pull request closed without merging**